python/dolfinx/fem/assemble.py

```diff
@@ -68,52 +106,64 @@ def create_matrix_nest(a: typing.List[typing.List[typing.Union[Form, cpp.fem.For
     return cpp.fem.create_matrix_nest(_create_cpp_form(a))
 
 
+Coefficients = collections.namedtuple('Coefficients', ['constants', 'coeffs'])
+
 # -- Scalar assembly ---------------------------------------------------------
 
 
-def assemble_scalar(M: typing.Union[Form, cpp.fem.Form]) -> PETSc.ScalarType:
-    """Assemble functional. The returned value is local and not accumulated
-    across processes.
+def assemble_scalar(M: typing.Union[Form, cpp.fem.Form],
+                    coeffs=Coefficients(None, None)) -> PETSc.ScalarType:
+    """Assemble functional. The returned value is local and not
```

Comment:
Should the documentation contain some hints on what shape or type the coefficients/constants should have, or do you think this is for expert usage only.

Reply:
I think that's more for experts, and it might change.